.github/workflows/default.yml

```diff
@@ -0,0 +1,289 @@
+name: Links main workflow
+
+on:
+  pull_request:
+    branches: master
+  push:
+    branches: master
+  schedule:
+    # Prime the caches every Monday
+    - cron: 0 1 * * MON
+
+jobs:
+  rule-check:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+        ocaml-compiler:
+          - 4.08.0
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - run: git diff-index --quiet HEAD --
+        shell: bash
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - run: opam install -y ocamlformat.0.14.2
+
+      - name: Ad-hoc rule check
+        run: |
+          eval $(opam config env)
+          make rule-check
+        shell: bash
+
+  build-doc:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get install -y python3-setuptools python3-pip
+          pip3 install -U Sphinx
+        shell: bash
+
+      - name: Compile documentation
+        run: make doc
+        shell: bash
+
+  interpreter:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+        ocaml-compiler:
+          - 4.08.0
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install system dependencies
+        run: sudo apt-get install -y libev-dev
+        shell: bash
+
+      - name: Use OCaml ${{ matrix.ocaml-compiler }}
+        uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - run: opam install -y conf-libev
+
+      - name: Install Links dependencies
+        run: >-
+          opam install
+          ./links.opam
+          --deps-only
+
+      - name: Build Links from source
+        run: |
+          eval $(opam env)
+          make all-ci
+        shell: bash
+
+      - name: Run interpreter testsuite
+        run: |
+         eval $(opam env)
+         make tests
+        shell: bash
+
+  postgres-mysql8-sqlite3:
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-20.04
+        ocaml-compiler:
+          - 4.08.0
+
+    runs-on: ${{ matrix.os }}
+
+    services:
+      postgres:
+        image: postgres:12
+        env:
+          POSTGRES_USER: links
+          POSTGRES_PASSWORD: links
+        ports:
+          - 5433:5432
+      mysql:
+        image: mysql:8.0
+        env:
+          MYSQL_USER: links
+          MYSQL_USER_PASSWORD: links
```

Comment:
I think if this becomes MYSQL_PASSWORD, the code to create the user shouldn't be necessary.

Reply:
It sadly is.

Reply:
These variables appear to be ignored by the container, and can probably be deleted.

Reply:
You still need to fix this otherwise the user account setup won't work.